porch/repository/pkg/git/git_test.go

```diff
@@ -284,3 +285,207 @@ func initRepo(repo *gogit.Repository) error {
 
 	return nil
 }
+
+const Kptfile = `
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: empty
+info:
+  description: Empty Package
+`
+
+func TestListPackagesEmpty(t *testing.T) {
+	testdata := TestDataAbs(t, "testdata")
+	tempdir := CreateTestTempDir(t)
+	tarfile := filepath.Join(testdata, "empty-repository.tar")
+	address := ServeGitRepository(t, tarfile, tempdir)
+
+	ctx := context.Background()
+	const (
+		repositoryName = "empty"
+		namespace      = "default"
+	)
+	var resolver repository.CredentialResolver
+
+	git, err := OpenRepository(ctx, repositoryName, namespace, &configapi.GitRepository{
+		Repo:      address,
+		Branch:    "main",
+		Directory: "/",
+		SecretRef: configapi.SecretRef{},
+	}, resolver, tempdir)
+	if err != nil {
+		t.Fatalf("Failed to open Git repository loaded from %q: %v", tarfile, err)
+	}
+
+	revisions, err := git.ListPackageRevisions(ctx)
+	if err != nil {
+		t.Fatalf("Failed to list packages from %q: %v", tarfile, err)
+	}
+	if got, want := len(revisions), 0; got != want {
+		t.Errorf("Number of packges in empty repository: got %d, want %d", got, want)
+	}
+
+	// TODO: Enable the rest of the test when gogit can push into an empty repository.
```

Comment:
I do think it can, it just needs us to do a little dance IIRC ... we can try to figure it out together!

Reply:
sure! I debugged into the go-git internals and it _seemed_ to me that the code is overly conservative. I'll find some time to dive deeper into this.